packages/core/test/scuttle.spec.js

```diff
@@ -0,0 +1,145 @@
+  )
+})
+
+test('scuttle - opts as object', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  const { all, configurables } = getPropsGroups(vmGlobalThis)
+  all.map((p) => vmGlobalThis[p])
+  scuttle(vmGlobalThis, {})
+  all.map((p) => vmGlobalThis[p])
+  scuttle(vmGlobalThis, { enabled: false })
+  all.map((p) => vmGlobalThis[p])
+  scuttle(vmGlobalThis, { enabled: true })
+  all.map((p) =>
+    !configurables.includes(p)
+      ? vmGlobalThis[p]
+      : t.throws(() => vmGlobalThis[p], { message: err(p) })
+  )
+})
+
+test('scuttle - exceptions', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  const { all, configurables } = getPropsGroups(vmGlobalThis)
+  all.map((p) => vmGlobalThis[p])
+  scuttle(vmGlobalThis, {
+    enabled: true,
+    exceptions: ['/[a-zA-Z0-9]*Array/', 'RegExp'],
+  })
+  const exceptions = [
+    'RegExp',
+    'Array',
+    'ArrayBuffer',
+    'Uint8Array',
+    'Int8Array',
+    'Uint16Array',
+    'Int16Array',
+    'Uint32Array',
+    'Int32Array',
+    'Float32Array',
+    'Float64Array',
+    'Uint8ClampedArray',
+    'BigUint64Array',
+    'BigInt64Array',
+    'SharedArrayBuffer',
+  ]
+  exceptions.map((p) => vmGlobalThis[p])
+  all
+    .filter((p) => !exceptions.includes(p))
+    .map(
+      (p) =>
+        configurables.includes(p) &&
+        t.throws(() => vmGlobalThis[p], { message: err(p) })
+    )
+})
+
+test('scuttle - scuttle func', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  const { all, configurables } = getPropsGroups(vmGlobalThis)
+  all.map((p) => vmGlobalThis[p])
+  let globalRef
+  const cb = (realm, scuttle) => ((globalRef = realm), scuttle(realm))
+  const opts = { enabled: true, scuttlerName: 'SCUTTLER' }
+  t.throws(() => scuttle(vmGlobalThis, opts), SCUTTLER_NAME_ERROR)
+  Object.defineProperty(vmGlobalThis, opts.scuttlerName, { value: cb })
+  scuttle(vmGlobalThis, opts)
+  t.is(
+    globalRef,
+    vmGlobalThis,
+    'expect global reference to be the same as provided by the scuttler function'
+  )
+  all.map((p) =>
+    !configurables.includes(p)
+      ? vmGlobalThis[p]
+      : t.throws(() => vmGlobalThis[p], { message: err(p) })
+  )
+})
+
+test('scuttle - resilient', (t) => {
+  const { vmGlobalThis } = evaluateWithSourceUrl('some-code', ';', {})
+  const { all, configurables } = getPropsGroups(vmGlobalThis)
+  all.map((p) => vmGlobalThis[p])
+  scuttle(vmGlobalThis, { enabled: true })
+  all.map((p) =>
+    !configurables.includes(p)
+      ? vmGlobalThis[p]
+      : t.throws(() => vmGlobalThis[p], { message: err(p) })
+  )
+  t.throws(() => Object.defineProperty(vmGlobalThis, 'Array', { value: 111 }), {
+    message: 'Cannot redefine property: Array',
+  })
+  vmGlobalThis.Array = 1
+  all.map((p) =>
+    !configurables.includes(p)
+      ? vmGlobalThis[p]
+      : t.throws(() => vmGlobalThis[p], { message: err(p) })
+  )
```

Comment:
```suggestion
  all.forEach((p) =>
    configurables.includes(p)
      ? t.throws(() => vmGlobalThis[p], { message: err(p) })
      : vmGlobalThis[p]
  )
```

Unless there's some subtle reason why this ternary condition should be inverted, in which case an inline comment is warranted.

Reply:
👍  to avoiding unnecessary negation in ternaries or if/else conditions

Reply:
a22476635